testkit/src/main/java/org/apache/calcite/sql/parser/SqlParserTest.java

```diff
@@ -5780,6 +5780,22 @@ private static Matcher<SqlNode> isCharLiteral(String s) {
             + "FROM `BIDS`");
   }
 
+  @Test void testWindowSpecExclusion() {
+    // parses correctly
+    sql("select sum(x) over (order by x rows between 2 preceding and 2 following exclude group) "
+        + "from emp")
+        .ok("SELECT (SUM(`X`) OVER (ORDER BY `X` ROWS BETWEEN 2 PRECEDING AND 2 "
+            + "FOLLOWING EXCLUDE GROUP))\n"
+            + "FROM `EMP`");
+    // doesn't parse without frame definition
+    sql("select sum(x) over (order by x ^exclude^ current row) from bids")
```

Comment:
Why do we need the `'^'` here?

Reply:
@asolimando this is a convention used to indicate the error position reported by the parser in the statement